web/workflow.R

```diff
@@ -11,184 +11,217 @@
-    && (PEcAn.utils::status.check("ADVANCED") == 0)) {
-  PEcAn.utils::status.start("ADVANCED")
-  q();
-}
-
-# Start ecosystem model runs
-if (PEcAn.utils::status.check("MODEL") == 0) {
-  PEcAn.utils::status.start("MODEL")
-  PEcAn.remote::runModule.start.model.runs(settings, stop.on.error = FALSE)
-  PEcAn.utils::status.end()
-}
-
-# Get results of model runs
-if (PEcAn.utils::status.check("OUTPUT") == 0) {
-  PEcAn.utils::status.start("OUTPUT")
-  runModule.get.results(settings)
-  PEcAn.utils::status.end()
-}
-
-# Run ensemble analysis on model output.
-if ("ensemble" %in% names(settings)
-    && PEcAn.utils::status.check("ENSEMBLE") == 0) {
-  PEcAn.utils::status.start("ENSEMBLE")
-  runModule.run.ensemble.analysis(settings, TRUE)
-  PEcAn.utils::status.end()
-}
-
-# Run sensitivity analysis and variance decomposition on model output
-if ("sensitivity.analysis" %in% names(settings)
-    && PEcAn.utils::status.check("SENSITIVITY") == 0) {
-  PEcAn.utils::status.start("SENSITIVITY")
-  runModule.run.sensitivity.analysis(settings)
-  PEcAn.utils::status.end()
-}
-
-# Run parameter data assimilation
-if ("assim.batch" %in% names(settings)) {
-  if (PEcAn.utils::status.check("PDA") == 0) {
-    PEcAn.utils::status.start("PDA")
-    settings <- PEcAn.assim.batch::runModule.assim.batch(settings)
-    PEcAn.utils::status.end()
-  }
-}
-
-# Run state data assimilation
-if ("state.data.assimilation" %in% names(settings)) {
-  if (PEcAn.utils::status.check("SDA") == 0) {
-    PEcAn.utils::status.start("SDA")
-    settings <- sda.enfk(settings)
-    PEcAn.utils::status.end()
-  }
-}
-
-# Run benchmarking
-if ("benchmarking" %in% names(settings)
-    && "benchmark" %in% names(settings$benchmarking)) {
-  PEcAn.utils::status.start("BENCHMARKING")
-  results <- papply(settings, function(x) calc_benchmark(x, bety))
-  PEcAn.utils::status.end()
-}
-
-# Pecan workflow complete
-if (PEcAn.utils::status.check("FINISHED") == 0) {
-  PEcAn.utils::status.start("FINISHED")
-  PEcAn.remote::kill.tunnel(settings)
-  db.query(paste("UPDATE workflows SET finished_at=NOW() WHERE id=",
-                 settings$workflow$id, "AND finished_at IS NULL"),
-           params = settings$database$bety)
-
-  # Send email if configured
-  if (!is.null(settings$email)
-      && !is.null(settings$email$to)
-      && (settings$email$to != "")) {
-    sendmail(settings$email$from, settings$email$to,
-             paste0("Workflow has finished executing at ", base::date()),
-             paste0("You can find the results on ", settings$email$url))
-  }
-  PEcAn.utils::status.end()
-}
-
-db.print.connections()
-print("---------- PEcAn Workflow Complete ----------")
+suppressMessages(library("PEcAn.all"))
+suppressMessages(library("PEcAn.utils"))
+suppressMessages(library("RCurl"))
+suppressMessages(library("optparse"))
+
+# --------------------------------------------------
+get_args <- function () {
+    option_list = list(
+      make_option(
```

Comment:
@robkooper should this use the `optparse::make_option()` syntax instead of loading libraries? (@kyclark this would eliminate the need to load the package above)
```suggestion
      optparse::make_option(
```